### DISCUSSION TITLE
Programming

### DISCUSSION THREAD

**Discussion by edwinkiran131** [community/community#17266] [Other Feature Feedback, Questions, & Ideas]
Hello,
i want to practice java and i am unable to find source code which could work in this,can anyone help me out

**Comment by cardison** [comment]
https://sonarcloud.io/github?gads_campaign=RoW1-Generic&gads_ad_group=Languages&gads_keyword=github%20java&gclid=CjwKCAjwtcCVBhA0EiwAT1fY7zKTzt7vdAy3xMmle3RbWy9GMFjH18xtCTbYuGeXg8ycySgL3R-dDhoCko8QAvD_BwE

this is a perfect website that you can find the source code,debugging and also learn about java

**Comment by LedjoLleshaj** [comment]
Best ways to learn java is by practising on small problem solving and writing project. i would suggest leetcode for problems.and a repository with the solutions of most of java problems.Dont cheat though,haha.Try doing them first! [Leetcode java solutions](https://github.com/varunu28/LeetCode-Java-Solutions)

And after getting better u will start to learn about different algorithms like sorting,search.... and a pretty cool repo with examples of these implementations is [in this link here](https://github.com/TheAlgorithms/Java/tree/master/src/main/java/com/thealgorithms)
Good luck and keep on coding!

**Comment by dendcode** [comment]
cloning spring pattern of project would help I think.

**Comment by d-coder111** [comment]
> Hello, i want to practice java and i am unable to find source code which could work in this,can anyone help me out

Hi edwinkiran131, Welcome to the Github Community👋

If you want to learn java from scratch, you can use following websites:
1) https://www.w3schools.com/java/
2) https://www.geeksforgeeks.org/java-tutorial/?ref=lbp

If you want to contribute to open source projects in java, you can use the following website:
https://www.oswc.is/search-projects

Hope u find it useful😍!!!

Let's connect and learn together and feel free to approach me if u need any help🤝😊.

Good luck👍🎉
